### PR TITLE
MCR-1579 MyCore REST API uses wrong base URL

### PR DESCRIPTION
**Pull request by eckstueck**
added new getBaseURL function that uses UriInfo to get the right base URL

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on e5e7ecab5b1ca9fb9fb436862f01dd7265bde667! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1221-pull_request-216

File: mycore-base/src/main/resources/config/mycore.properties [hunk 1]
```diff
@@ -398,6 +398,7 @@ MCR.ConfigurationDirectory.template.files=mycore.properties,resources/META-INF/p
 ##############################################################################
 
 MCR.Jersey.resource.packages=org.mycore.frontend.jersey.resources,org.mycore.services.queuedjob
+MCR.Jersey.resource.urlPattern=rsc
```
Comment (yagee-de) [inline mycore-base/src/main/resources/config/mycore.properties]: urlPattern is misleading and "resource" needs to be with capital letter 'R'. Suggest: `MCR.Jersey.Resource.ApplicationPaths=rsc`

File: mycore-base/src/main/java/org/mycore/frontend/jersey/MCRJerseyUtil.java [hunk 2]
```diff
@@ -146,4 +148,22 @@ public static String fromStatusCode(int statusCode) {
...
+     *
+     * @param info the UriInfo
+     *
+     * @return base URL of the mycore system as string
+     */
+    public static String getBaseURL(UriInfo info) {
+        String baseURL = info.getBaseUri().toString();
+        String urlPattern = MCRConfiguration.instance().getString("MCR.Jersey.resource.urlPattern");
+        for(String pattern : urlPattern.split(",")){
+            if (baseURL.endsWith(pattern + "/")){
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/jersey/MCRJerseyUtil.java]: you also need to to add `'/'` at the beginning of the path to make it safe

File: mycore-base/src/main/java/org/mycore/frontend/jersey/MCRJerseyUtil.java [hunk 3]
```diff
@@ -146,4 +148,22 @@ public static String fromStatusCode(int statusCode) {
...
+     * Returns the base URL of the mycore system.
+     *
+     * @param info the UriInfo
+     *
+     * @return base URL of the mycore system as string
+     */
+    public static String getBaseURL(UriInfo info) {
+        String baseURL = info.getBaseUri().toString();
+        String urlPattern = MCRConfiguration.instance().getString("MCR.Jersey.resource.urlPattern");
+        for(String pattern : urlPattern.split(",")){
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/jersey/MCRJerseyUtil.java]: use `MCRConfiguration.getStrings(String)` to get a List of values

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Review by yagee-de — APPROVED** [review]

**Pull request merged**